packages/cojson-storage-indexeddb/src/idbClient.ts

```diff
@@ -220,7 +249,15 @@ export class IDBClient implements DBClientInterface {
     );
   }
 
-  async unitOfWork(operationsCallback: () => unknown[]) {
-    return Promise.all(operationsCallback());
+  async transaction(operationsCallback: () => unknown) {
+    const txEntry = this.startTransaction();
+    try {
+      await operationsCallback();
+      this.closeTransaction(txEntry);
+    } catch (error) {
+      txEntry.tx.abort();
+      this.closeTransaction(txEntry);
+      throw error;
+    }
```

Comment:
I mean to improve this code, the main idea was to bulk operations in the same transactions only when declared.